crypto/src/lib.rs

```diff
@@ -175,7 +173,6 @@ impl KeyPair {
     ///
     /// # Errors
     /// Fails if decoding fails
-    #[cfg(any(feature = "std", feature = "ffi_import"))]
     pub fn generate() -> Result<Self, Error> {
```

Comment:
Maybe this should be gated behind some `cfg` to make sure it isn't available in wasm? Currently it will just panic when the stub `getrandom` impl will get called and panic.

Maybe also add a note in doc that this won't work because randomness in smart contracts is not a thing

Reply:
My suggestion here is to guard it behind a feature flag and make that feature part of the `default` feature set. Since smart contracts always use `default-features = false` this will remove friction from all users